### DISCUSSION TITLE
.keys() does not tell me the missing keys

### DISCUSSION THREAD

**Discussion by pihentagy** [chaijs/chai#1480] [Support] [answered]
Just started to use chai via postman tests.

I wonder if I can get chai to print the missing keys.

For the following test

```js
pm.test("Have all mentioned fiels", () => {
    const obj = {"hello": "world"};
    pm.expect(obj).to.have.all.keys("hello", "goodbye");
})
```

The result is: 
```plaintext
Have all mentioned fiels | AssertionError: expected { hello: 'world' } to have keys 'hello', and 'goodbye'
```

Which is correct, but does not tell me: 'hey, goodbye is the key, which is missing.

Any way to get chai print just the missing keys?

**Comment by keithamus** [comment, marked as answer]
Right now this is not possible, it would require us to change the error message generated (which is definitely possible)! It's something we're aware of and are working on.
